File: mobile/lib/features/leads/domain/lead.dart
class Lead {
  const Lead({
    required this.id,
    required this.fullName,
    required this.phone,
    required this.email,
    required this.source,
    required this.status,
    required this.budgetMin,
    required this.budgetMax,
    required this.preferredLocation,
    required this.propertyType,
    required this.notes,
  });

  factory Lead.fromJson(Map<String, dynamic> json) => Lead(
    id: json['id'] as String,
    fullName: json['fullName'] as String,
    phone: json['phone'] as String?,
    email: json['email'] as String?,
    source: json['source'] as String,
    status: json['status'] as String,
    budgetMin: (json['budgetMin'] as num?)?.toDouble(),
    budgetMax: (json['budgetMax'] as num?)?.toDouble(),
    preferredLocation: json['preferredLocation'] as String?,
    propertyType: json['propertyType'] as String?,
    notes: json['notes'] as String?,
  );

  final String id;
  final String fullName;
  final String? phone;
  final String? email;
  final String source;
  final String status;
  final double? budgetMin;
  final double? budgetMax;
  final String? preferredLocation;
  final String? propertyType;
  final String? notes;
}

File: client/real-estate-crm-react/src/types/lead.ts
export const LeadStatus = {
  New: "New",
  Contacted: "Contacted",
  Interested: "Interested",
  Viewing: "Viewing",
  Negotiation: "Negotiation",
  Reserved: "Reserved",
  Contracted: "Contracted",
  Lost: "Lost",
} as const;
export type LeadStatus = (typeof LeadStatus)[keyof typeof LeadStatus];

export const LeadSource = {
  Website: "Website",
  Facebook: "Facebook",
  Instagram: "Instagram",
  Google: "Google",
  Referral: "Referral",
  WalkIn: "WalkIn",
  Phone: "Phone",
  Other: "Other",
} as const;
export type LeadSource = (typeof LeadSource)[keyof typeof LeadSource];

export const LeadActivityType = {
  Call: "Call",
  WhatsApp: "WhatsApp",
  Email: "Email",
  Meeting: "Meeting",
  Viewing: "Viewing",
  Note: "Note",
  FollowUp: "FollowUp",
} as const;
export type LeadActivityType = (typeof LeadActivityType)[keyof typeof LeadActivityType];

export interface Lead {
  id: string;
  fullName: string;
  phone: string | null;
  email: string | null;
  source: LeadSource;
  status: LeadStatus;
  budgetMin: number | null;
  budgetMax: number | null;
  preferredLocation: string | null;
  propertyType: string | null;
  assignedAgentId: string | null;
  notes: string | null;
  createdAt: string;
  updatedAt: string;
}

export interface CreateLeadRequest {
  fullName: string;
  phone?: string | null;
  email?: string | null;
  source: LeadSource;
  budgetMin?: number | null;
  budgetMax?: number | null;
  preferredLocation?: string | null;
  propertyType?: string | null;
  assignedAgentId?: string | null;
  notes?: string | null;
}

export interface UpdateLeadRequest extends CreateLeadRequest {
  status: LeadStatus;
}

export interface LeadListQuery {
  page?: number;
  pageSize?: number;
  search?: string;
  status?: LeadStatus;
  assignedAgentId?: string;
  source?: LeadSource;
  sortBy?: string;
  sortDirection?: "asc" | "desc";
}

export interface AssignLeadRequest {
  agentId: string;
}

export interface LeadActivity {
  id: string;
  leadId: string;
  userId: string;
  type: LeadActivityType;
  description: string | null;
  activityDate: string;
  createdAt: string;
}

export interface CreateLeadActivityRequest {
  type: LeadActivityType;
  description?: string | null;
  activityDate?: string | null;
}
